core/internal/testutils/pgtest/txdb.go

```diff
@@ -103,6 +102,9 @@ func (d *txDriver) deleteConn(c *conn) error {
 	d.Lock()
 	defer d.Unlock()
 
+	if d.conns[c.dsn] != c {
+		return nil // already been replaced
+	}
```

Comment:
Ensure we haven't been replaced already.

Reply:
Is this just an optimisation? i.e. will it still work if this is removed?

Reply:
No, it is necessary to prevent trampling an existing one.